plugins/SpectrumAnalyzer/SaProcessor.h

```diff
@@ -0,0 +1,121 @@
+/* SaProcessor.h - declaration of SaProcessor class.
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * Based partially on Eq plugin code,
+ * Copyright (c) 2014 David French <dave/dot/french3/at/googlemail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SAPROCESSOR_H
+#define SAPROCESSOR_H
+
+#include <QImage>
+#include <QMutex>
+#include <vector>
+
+#include "fft_helpers.h"
+#include "SaControls.h"
```

Comment:
Then I couldn't use e.g. `m_controls->m_pauseModel.value()` in the cpp file, I think? So I would have to move the include to the .cpp file as well, so it would be forward decl. in .h and include in .cpp; again, seems more complicated for no benefit?